# PythonBasic习题

## 进制练习
- 计算 0xFFFF 表示多少？
- 计算 0b1001 表示多少？
- 计算 0o71  表示多少？
- 把 13 转为2进制。
- 把 18 转为16进制。

In [10]:
print("0xFFFF表示：15*16**3+15*16**2+15*16+15 =", 15*16**3+15*16**2+15*16+15)
print("0b1001表示：1*2**3 + 0 + 0 + 1*2**0 =", 15*16**3+15*16**2+15*16+15)
print("0o71表示：7*8**2 + 1 =", 7*8**1 + 1)
print("13的二进制：13/2 = 6...1 -> 6/2 = 3...0 -> 3/2 = 1...1 -> 1 ->", bin(13))
print("18的十六进制：18/16 = 1...2 ->", hex(18))

0xFFFF表示：15*16**3+15*16**2+15*16+15 = 65535
0b1001表示：1*2**3 + 0 + 0 + 1*2**0 = 65535
0o71表示：7*8**2 + 1 = 57
13的二进制：13/2 = 6...1 -> 6/2 = 3...0 -> 3/2 = 1...1 -> 1 -> 0b1101
18的十六进制：18/16 = 1...2 -> 0x12


## 简单计算
- 一个商店卖西瓜7元一个，你带20元能买几个西瓜，找零多少钱？
- 一个学生毕业薪资10000元，每年涨20%，十年后他的薪资是多少？
- 已知矩形的长边长6cm，短边长4cm，写出程序求出矩形的周长和面积？
- 中国古代的称是16两一斤，请问216两是古代的几斤几两？
- 从凌晨0:0:0计时，现在已经过了63320秒，请问现在是几时几分几秒（可以用地板除和求余实现）？

In [30]:
print("找零", 20 % 7, "元")
print("十年后他的薪资为", 10000 * (1 + 0.2) ** 10, "元")
print("矩形的周长为{}厘米，面积为{}平方厘米".format((6 + 4) * 2, 6 * 4))
print("216两相当于古代的{}斤{}两".format(216 // 16,216 % 16))
shi = 63320//(60*60)
fen = 63320%(60*60)//60
miao = 63320%(60*60)%60
print("现在是{}时{}分{}秒".format(shi, fen, miao))

找零 6 元
十年后他的薪资为 61917.364223999975 元
矩形的周长为20厘米，面积为24平方厘米
216两相当于古代的13斤8两
现在是17时35分20秒


## statement 练习
指定一个圆的半径r为3cm
- 计算出圆的周长
- 计算出圆的面积
    - 要求用变量代替3cm和圆周率3.1415926

In [23]:
r = 3
pi = 3.1415926
c = 2 * pi * r
s = pi * r ** 2
print("圆的周长为{}厘米，圆的面积为{}平方厘米".format(c, s))

圆的周长为18.849555600000002厘米，圆的面积为28.2743334平方厘米


## 精度练习
- 将数字3.14用变量pi绑定
- 将pi变量转为整数，用变量i绑定
- 将pi变量与i变量相减，结果用f绑定
- 判断f是否等于0.14？为什么？
- 删除所有的变

In [32]:
pi = 3.14
i = int(pi)
f = pi - i
print(f, "这是因为损失了精度")
del f, i, pi

0.14000000000000012 这是因为损失了精度


## input 练习
- 分三次输入如当前的小时，分钟，秒钟，在终端打印已距离凌晨0:0:0过了多少秒？

In [33]:
shi = int(input("请输入小时："))
fen = int(input("请输入分钟："))
miao = int(input("请输入秒钟："))
print("距离凌晨过了{}秒".format(shi*3600 + fen*60 + miao))

请输入小时：15
请输入分钟：28
请输入秒钟：45
距离凌晨过了55725秒


## if 练习1

- 请输入一个数字，判断这个数是0、正数、还是负数

In [7]:
n = int(input("请输入一个数字，判断这个数是0、正数、负数："))
if n > 0:
    print("正数")
elif n < 0:
    print("负数")
else:
    print("等于零")

请输入一个数字，判断这个数是0、正数、负数：16
正数


## if 练习2

- 输入一个季度1~4输出这个季度有那几个月、如果输入不是1~4的数，提示用户您的输入有误！
- 输入一年中的月份（1~12），输出这个月在哪几个季度，如果输入的是其他数，提示您输入的有误！

In [8]:
s = int(input("请输入1~4个季度"))
if s == 1:
	print("第一季度有1、2、3个月")
elif s == 2:
	print("第二季度有4、5、6个月")
elif s == 3:
	print("第三季度有7、8、9个月")
elif s == 4:
	print("第四季度有10、11、12个月")
else:
	print("您输入的季度不合法")

m = int(input("请输入1~12月份"))
if 1 <= m <= 3:
	print("第一季度")
elif 4 <= m <= 6:
	print("第二季度")
elif 7 <= m <= 9:
	print("第三季度")
elif 10 <= n <= 12:
	print("第四季度")
else:
	print("您输入的月份不合法")

请输入1~4个季度3
第三季度有7、8、9个月
请输入1~12月份7
第三季度


## if 嵌套练习

- 输入一个学生的成绩（0-100）
- 判断这个学生的成绩是优（90-100）、良（80-89）、及格（60-79）、不及格、不合法

In [9]:
n = int(input("请输入学生成绩"))
if 0 <= n <= 100:
    if 90 <= n <= 100:
        print("优秀")
    elif 80 <= n < 90:
        print("良好")
    elif 60 <= n < 80:
        print("及格")
    else:
        print("不及格")
else:
    print("您输入的成绩不合法！")

请输入学生成绩75
及格


## 条件表达式练习
- 用if语句计算出这个数的绝对值并打印出来
- 用条件表达式计算出这个数的绝对值并打印出来

In [12]:
n = int(input("请输入整数："))
n = -n if n < 0 else n
print(n)

请输入整数：-5
5


## 闰年

In [13]:
y = int(input("请输入年份："))
if y % 400 == 0 or (y % 4 == 0 and y % 100 != 0):
	print("为闰年")
else: print("不是闰年")

请输入年份：2004
为闰年


## BMI

In [18]:
h = int(input("请输入您的身高"))
w = int(input("请输入您的体重"))
BMI = w/(h/100)**2
print("您的BMI为", BMI)
if BMI < 18.5:
	print("过轻")
elif 18.5 <= BMI <= 24:
	print("正常")
else:
	print("超重")

请输入您的身高178
请输入您的体重75
您的BMI为 23.671253629592222
正常


## 字符串练习
- 写一个程序，打印高度为4的矩形方框，要求输入一个整数，代表矩形的宽度，输出矩形

In [19]:
n = int(input("请输入宽度："))
print("*"*n)
print('*'+' '*(n-2)+'*')
print('*'+' '*(n-2)+'*')
print("*"*n)

请输入宽度：10
**********
*        *
*        *
**********


## 切片练习
1. 写一个程序，输入一个字符串，把字符串的第一个和最后一个字符去掉，打印出处理后的字符串。
2. 输入任意一个字符串，判断这个字符串是否是回文，回文是指中心对称的文字。

In [21]:
s = input("请输入一串字符串")
print(s[1:-1])
s = input("请输入一串字符串，判断是否为回文")
if s == s[::-1]:
    print("是回文")
else:
    print("不是回文")

请输入一串字符串haoenwei
aoenwe
请输入一串字符串，判断是否为回文abcba
是回文


## 字符练习1
1. 写一个程序，输入一段字符串，如果字符串不为空，则把第一个字符串的编码打印出来
2. 写一个程序，输入一个整数值（0～65535），打印这个数值对应的字符。

In [5]:
s = input("请输入一串字符串，打印出第一个字符的编码")
print(ord(s[0]))
c = int(input("请输入一个整数值(0~65536)打印出这个数值对应的字符"))
print(chr(c))

请输入一串字符串，打印出第一个字符的编码hello
104
请输入一个整数值(0~65536)打印出这个数值对应的字符20013
中


## 字符练习
1. 用字符串*运算符打印三角形，要求输入一个整数，此整数代表三角形离左侧的字节数。
2. 输入三行文字，让这三行文字在一个方框中显示，如输入（不要输入中文）：

In [10]:
n = int(input("请输入三角形离左侧的字节数："))
print(' '*n + '  *  ')
print(' '*n + ' *** ')
print(' '*n + '*****')

s1 = input("请输入第一行文字")
s2 = input("请输入第二行文字")
s3 = input("请输入第三行文字")

m = len(s1)
if len(s2) > m:
	m = len(s2)
if len(s3) > m:
	m = len(s3)

line1 = "+" + m * '-' + "+"
line2 = "|" + round((m-len(s1))/2) * " " + s1 + round((m-len(s1))/2) * " "+"|"
line3 = "|" + round((m-len(s2))/2) * " " + s2 + round((m-len(s2))/2) * " "+"|"
line4 = "|" + round((m-len(s3))/2) * " " + s3 + round((m-len(s3))/2) * " "+"|"
print(line1)
print(line2)
print(line3)
print(line4)
print(line1)

请输入三角形离左侧的字节数：5
       *  
      *** 
     *****
请输入第一行文字asdfsdf
请输入第二行文字sadf
请输入第三行文字gdfgsdfgdfg
+-----------+
|  asdfsdf  |
|    sadf    |
|gdfgsdfgdfg|
+-----------+


## 字符练习
1. 判断您输入的字符串有几个空格
2. 将原字符串的左右空白字符去掉，打印出有效字符的长度。
3. 判断您输入的是否为数字。

In [16]:
s = input("请输入一串字符串，判断有几个空格")
print("有%d个空格"%s.count(' '))

s = input("请输入字符串，去掉左右空白字符，打印出有效字符长度")
j = 0
for i in s.strip():
    j += 1
print(s.strip(), "有%d个有效字符" % j)

s = input("请输入字符串，判断是否为数字")
if s.isdigit():
	print("是数字")
else:
	print("不是数字")

请输入一串字符串，判断有几个空格  hello
有2个空格
请输入字符串，去掉左右空白字符，打印出有效字符长度haha    
haha 有4个有效字符
请输入字符串，判断是否为数字6
是数字


## 格式字符练习
1. 输入三行文字，让这些文字依次以20字符的宽度右对齐，输出。
2. 能否以最长字符串的长度进行右对齐显示，左侧填充空格。

In [18]:
s1 = input("请输入第一行字符串")
s2 = input("请输入第二行字符串")
s2 = input("请输入第三行字符串")
print("%20s" % s1)
print("%20s" % s2)
print("%20s" % s3)

print("以最长的字符串为长度进行右对齐显示")
m = len(s1)
if len(s2) > m:
	m = len(s2)
if len(s3) > m:
	m = len(s3)

# 或 m = max(len(s1),len(s2),len(s3))
fmt = "%%%ds" % m
print(fmt % s1)
print(fmt % s2)
print(fmt % s3)

请输入第一行字符串hello
请输入第二行字符串world
请输入第三行字符串hahahaha
               hello
            hahahaha
         gdfgsdfgdfg
以最长的字符串为长度进行右对齐显示
      hello
   hahahaha
gdfgsdfgdfg


## whlie 练习1
1. 输入一个整数，用end变量绑定，打印出1～end的所有整数，包含end。
2. 写程序，输入第二个整数，第一个用begin绑定，第二个用end绑定，打印出begin～end的所有整数。

In [30]:
n = int(input("请输入End："))
i = 1
while i <= n:
    print(i)
    i += 1

请输入End：5
1
2
3
4
5
